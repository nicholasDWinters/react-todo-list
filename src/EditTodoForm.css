.EditTodoForm {
    margin-top: 10px;
}

.EditTodoForm-label, .EditTodoForm-button, .EditTodoForm-input {
    margin-left: 10px;
}