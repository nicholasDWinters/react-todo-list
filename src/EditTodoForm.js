import React, { useState } from "react";
import './EditTodoForm.css';

const EditTodoForm = ({ editTodo, todo, resetHidden, rendered }) => {
    const INITIAL_STATE = {
        id: todo.id,
        task: todo.task
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData({
            ...formData,
            [name]: value
        })
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        editTodo({ ...formData });
        resetHidden();
    }

    return (
        <form onSubmit={handleSubmit} className='EditTodoForm' style={rendered}>
            <label className='EditTodoForm-lable' htmlFor="todo">Edit</label>
            <input
                className='EditTodoForm-input'
                id="todo"
                type="text"
                name="task"
                placeholder="Edit Todo"
                value={formData.task}
                onChange={handleChange}
            />
            <button className='EditTodoForm-button' >Edit Todo</button>
        </form>
    )

}

export default EditTodoForm;