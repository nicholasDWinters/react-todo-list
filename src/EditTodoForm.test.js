import React from "react";
import { render } from "@testing-library/react";
import EditTodoForm from './EditTodoForm';

const todo = {
    task: 'shop',
    id: 1
}

it("renders without crashing", function () {
    render(<EditTodoForm todo={todo} />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<EditTodoForm todo={todo} />);
    expect(asFragment()).toMatchSnapshot();
});
