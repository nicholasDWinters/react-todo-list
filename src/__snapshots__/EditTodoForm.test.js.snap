// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="EditTodoForm"
  >
    <label
      class="EditTodoForm-lable"
      for="todo"
    >
      Edit
    </label>
    <input
      class="EditTodoForm-input"
      id="todo"
      name="task"
      placeholder="Edit Todo"
      type="text"
      value="shop"
    />
    <button
      class="EditTodoForm-button"
    >
      Edit Todo
    </button>
  </form>
</DocumentFragment>
`;
